src/canvas/Geometry/Curve.tsx

```diff
@@ -66,6 +67,22 @@ export abstract class Curve extends Segment {
         }
     };
 
+    /* 
+     * Checks if a point is near the curve by computing a number of points on taht curve 
+     * and then checking if the given point is within a radius at the given threshold of 
+     * each computed point.
+    */
+    isPointNearSegment = (point: Point, threshold: number): Point | null => {
```

Comment:
Hmm, that name is more confusing to me. I liked isPointNearSegment because you can treat is a returning a boolean, if the point is not near the segment it will return null, which is falsey, and if it is near the segment, it will return the closest point which is truthy.

Reply:
As we agreed, I will keep the same name and update the comment for clarity.